### ISSUE TITLE
Permission denied when using persistent volumes

### ISSUE THREAD

**Issue by omri-shilton** [8gears/n8n-helm-chart#71]
chart version = 0.20.1
when using
```
persistence:
  enabled: true
  type: dynamic

```
deploying this for the first time getting this error 
```
Error: EACCES: permission denied, open '/home/node/.n8n/config'
    Code: EACCES

```

Running on EKS

**Comment by Vad1mo** [comment]
The pvc is mounted to /home/node/.n8n/ ([here](https://github.com/8gears/n8n-helm-chart/blob/6c9995a387b3286432cbafb730639b1b337426a8/templates/deployment.yaml#L78C104-L78C119)) 

and n8n assumes the dir `config` exists, but It is not there. 

You have two options as a workaround:
1. Create that dir `config` somehow, manually.
2. set ENV var `N8N_CONFIG_FILES= /n8n-config/config.json,/n8n-secret/secret.json,/home/node/.n8n/ `

I will find a solution to that problem. The only options that jumps into my mind right now is init-container. 

Happy to accept PRs


Btw, why do you need a volume?

**Comment by omri-shilton** [comment]
> Btw, why do you need a volume?

Im using a community node for 'AWS Secret Manager', and installing it via the UI writes it to the home directory of N8N, so i wanted to have the persistent.

I will try the workaround

Thanks for the quick response

**Comment by Vad1mo** [comment]
ahh makes sense, thanks

**Comment by omri-shilton** [comment]
I tried the second workaround and it didnt work.. 
I dont think there is a way to do the first one..

**Comment by szechp** [comment]
having the same error, "Create that dir config somehow, manually" will not work in k8s, since i cant acces the pod when it fails.

**Comment by szechp** [comment]
quick and dirty fix:

add this to your values.yaml:
```yaml
  securityContext:
    runAsUser: 0
    runAsGroup: 0
```
**this will run the container as root!**

this then propagates the root user security context in the container (see [here](https://github.com/8gears/n8n-helm-chart/blob/6c9995a387b3286432cbafb730639b1b337426a8/templates/deployment.yaml#L55) and makes the config accessible again. 

this is not a good thing, i actually don't want to run my containers in k8s as root. but its the only way i to get persistency running rn.

**Comment by omri-shilton** [comment]
@Vad1mo any news regarding a proper fix for this?

**Comment by Vad1mo** [comment]
fixed security context  in 0.22, not sure if that fixes your issue. We might need to have another fix

https://artifacthub.io/packages/helm/open-8gears/n8n?modal=changelog

**Comment by Vad1mo** [comment]
do you see still the issue with 0.22 @aep and @omri-shilton ?

**Comment by aep** [comment]
Yes. Copying the config (not a dir!) manually works, although it's super annoying.

**Comment by Vad1mo** [comment]
I don't have time this week to implement a fix. But I am happy to accept any PRs coming that ways .

What you basically need to do is create an init container. I would use the same n8n container and add a command that will overwrite the entrypoint. 

the command should be `mkdir -p /home/node/.n8n/  `

**Comment by mfbrj** [comment]
For me, only worked with these values (chart 1.0.2):

```yaml
main:

...

  podSecurityContext:
    runAsNonRoot: false
    runAsUser: 1000
    runAsGroup: 1000
    fsGroup: 1000

  initContainers:
  - name: init-data-dir
    image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
    command: [ "/bin/sh", "-c", "chown 1000:1000 /home/node/.n8n" ]
    volumeMounts:
      - name: data
        mountPath: /home/node/.n8n
    securityContext:
      runAsUser: 0
      runAsGroup: 0

...


```

**Comment by Vad1mo** [comment]
I'll try to make an example and see if that works.

what is your PVC set up or infra look like?

**Comment by mfbrj** [comment]
The install its on a managed k8s cluster on Digital Ocean.

I'm using:
1 n8n
1 worker
1 webhook
1 valkey
1 postgres

The helm install created only 1 pvc for all 3 n8n pods (n8n, worker and webhook), despite I configured `persistence` on each tier. Then I needed to install a NFS storageclass to use accessModes `ReadWriteMany`.

It is also worth noting that the initContainer [example](https://github.com/8gears/n8n-helm-chart/blob/548fab3e94039de2269cadba4e8d98184d66d821/charts/n8n/values.yaml#L226) in values.yml was quite unfortunate, since the example suggests that creating the `/home/node/.n8n` directory is the solution. However, such a directory is already created automatically when the volume is mounted. And, in my case, the directory was always created with permissions for user 0 (root) even though I used fsGroup 1000.

> I'll try to make an example and see if that works.
> 
> what is your PVC set up or infra look like?
